src/blenderbim/blenderbim/bim/module/qto/operator.py

```diff
@@ -150,3 +150,31 @@ def _execute(self, context):
             ifcopenshell.api.run("pset.edit_qto", self.file, qto=qto, properties={props.prop_name: result})
             PsetData.load(self.file, obj.BIMObjectProperties.ifc_definition_id)
         return {"FINISHED"}
+
+class AssignPsetQto(bpy.types.Operator):
+    bl_idname = "bim.assign_pset_qto"
+    bl_label = "Assign Pset Qto"
+    bl_options = {"REGISTER", "UNDO"}
+    bl_description = "Assign pset qto to selected object"
+
+    @classmethod
+    def poll(cls, context):
+        return context.active_object and context.selected_objects
+
+    def execute(self, context):
```

Comment:
Shouldn't this use the new _execute() function? Otherwise undo won't work. Same goes for the other operator.

Reply:
Yep, done! I'll do a PR with this and other fixes.